### ISSUE TITLE
Regression with `unused_imports` on macro used in documentation

### ISSUE THREAD

**Issue by dylni** [rust-lang/rust#148143]
The `unused_imports` lint is triggered on nightly when it is not on beta or stable. The import is required.

### Code

I tried this code:

```rust
//! See:
#![doc = a!()]

macro_rules! a {
    () => { "Documentation" };
}
use a;

fn main() {}
```

I expected to see this happen: *No warnings*

Instead, this happened:

```
warning: unused import: `a`
 --> src/main.rs:7:5
  |
7 | use a;
  |     ^
  |
  = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default
```

<https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=c16575b4d9f265e4e6b9394ca8d22760>

### Version it worked on

It most recently worked on: 1.91.0-beta.10

### Version with regression

`rustc --version` (differs from beta by two versions even after updating):
```
rustc 1.93.0-nightly (34f954f9b 2025-10-25)
```

@rustbot modify labels: +regression-from-stable-to-nightly -regression-untriaged

**Comment by madsmtm** [comment]
Bisected to e1318422222101423fdbbe0f375109c3b40f75bc

**Comment by madsmtm** [comment]
Assuming that's the PR, it had a crater run with fairly minimal impact, so I'll go with P-medium
